src/github.com/matrix-org/dendrite/roomserver/api/query.go

```diff
@@ -104,6 +104,25 @@ type QueryEventsByIDResponse struct {
 	Events []gomatrixserverlib.Event `json:"events"`
 }
 
+// QueryMembershipForUserRequest is a request to QueryMembership
+type QueryMembershipForUserRequest struct {
+	// ID of the room to fetch membership from
+	RoomID string `json:"room_id"`
+	// ID of the user sending the request
+	Sender string `json:"sender"`
```

Comment:
Should this be changed from sender to UserID ?
since we want the membership for this user and not the sender.
@erikjohnston

Reply:
Yeah, I think so